types/react/next.d.ts

```diff
@@ -169,4 +169,31 @@ declare module '.' {
     };
 
     export function unstable_useOpaqueIdentifier(): OpaqueIdentifier;
+
+    /**
+     * this should be an internal type
+     */
+    interface MutableSource<T> {
+        _source: T;
+    }
+
+    export type MutableSourceSubscribe<T> = (source: T, callback: () => void) => () => void;
+
+    /**
+     * @param source A source could be anything as long as they can be subscribed to and have a "version".
+     * @param getVersion A function returns a value which will change whenever part of the source changes.
+     */
+    export function unstable_createMutableSource<T>(source: T, getVersion: () => any): MutableSource<T>;
```

Comment:
```suggestion
    export function unstable_createMutableSource<T>(source: T, getVersion: () => unknown): MutableSource<T>;
```
Newer types can use `unknown`.

Reply:
Callbacks are contravariant so this needs to be `() => any`